### PR TITLE
[semi-modular] thieving gloves

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Because Cobalt didn't lock the bounty apparently when Zephyr asked Cobalt, or Zephyr did it after I already reacted, and Cobalt didn't check if someone had claimed it. 

Thieving gloves: 

Stripping pocket and ID slots without a message and without dropping the thing you are taking onto the ground and placing it directly into your hand instead.

*from the bounty and unimplementable (would have to be upstream and they would deny it): stripping only one item at a time*

## How This Contributes To The Skyrat Roleplay Experience

Bounty

## Changelog

:cl:
add: adds thieving gloves
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
Hmm yes, steal my code almost entirely with very few changes. Interesting idea.

**Comment by YakumoChen** [comment]
Doesn't use the dreaded glove microchips so I already like this one more B)

File: code/datums/elements/strippable.dm [hunk 1]
```diff
@@ -148,18 +151,20 @@
...
-		span_userdanger("[user] tries to remove your [item.name]."),
-		ignored_mobs = user,
-	)
+	var/is_silent = can_be_silent && HAS_TRAIT(user, TRAIT_STICKY_FINGERS) //SKYRAT EDIT ADDITION
+	if (!is_silent)
+		source.visible_message(
+			span_warning("[user] tries to remove [source]'s [item.name]."),
+			span_userdanger("[user] tries to remove your [item.name]]."),
+			ignored_mobs = user,
+		)
```
Comment (GoldenAlpharex) [inline code/datums/elements/strippable.dm]:
```suggestion
	// SKYRAT EDIT ADDITION START - Thieving gloves
	var/is_silent = can_be_silent && HAS_TRAIT(user, TRAIT_STICKY_FINGERS)
	if (!is_silent)
		source.visible_message(
			span_warning("[user] tries to remove [source]'s [item.name]."),
			span_userdanger("[user] tries to remove your [item.name]]."),
			ignored_mobs = user,
		)
	// SKYRAT EDIT ADDITION END
```

File: code/datums/elements/strippable.dm [hunk 2]
```diff
@@ -291,7 +296,9 @@
 
 /// A utility function for `/datum/strippable_item`s to start unequipping an item from a mob.
 /proc/start_unequip_mob(obj/item/item, mob/source, mob/user, strip_delay)
-	if (!do_mob(user, source, strip_delay || item.strip_delay, interaction_key = REF(item)))
+	//SKYRAT EDIT ADDITION -
```
Comment (GoldenAlpharex) [inline code/datums/elements/strippable.dm]:
```suggestion
	//SKYRAT EDIT ADDITION - Thieving gloves
```

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -1049,7 +1049,15 @@ attack_basic_mob
 	return !HAS_TRAIT(src, TRAIT_NODROP) && !(item_flags & ABSTRACT)
 
 /obj/item/proc/doStrip(mob/stripper, mob/owner)
-	return owner.dropItemToGround(src)
+	//SKYRAT EDIT BEGIN - original  return owner.dropItemToGround(src)
```
Comment (GoldenAlpharex) [inline code/game/objects/items.dm]:
```suggestion
	//SKYRAT EDIT BEGIN - Thieving gloves - Original:  return owner.dropItemToGround(src)
```

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -1074,13 +1074,17 @@
 	if(!what.canStrip(who))
 		to_chat(src, span_warning("You can't remove \the [what.name], it appears to be stuck!"))
 		return
-	who.visible_message(span_warning("[src] tries to remove [who]'s [what.name]."), \
-					span_userdanger("[src] tries to remove your [what.name]."), null, null, src)
+	var/silent = HAS_TRAIT(who, TRAIT_STICKY_FINGERS) && (where & (ITEM_SLOT_ID|ITEM_SLOT_LPOCKET|ITEM_SLOT_RPOCKET)) //SKYRAT EDIT ADDITION
+	if (!silent)
+		who.visible_message(span_warning("[src] tries to remove [who]'s [what.name]."), \
+						span_userdanger("[src] tries to remove your [what.name]."), null, null, src)
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/living.dm]:
```suggestion
	// SKYRAT EDIT ADDITION START - Thieving gloves
	var/silent = HAS_TRAIT(who, TRAIT_STICKY_FINGERS) && (where & (ITEM_SLOT_ID|ITEM_SLOT_LPOCKET|ITEM_SLOT_RPOCKET))
	if (!silent)
		who.visible_message(span_warning("[src] tries to remove [who]'s [what.name]."), \
						span_userdanger("[src] tries to remove your [what.name]."), null, null, src)
	// SKYRAT EDIT ADDITION END
```

File: code/modules/mob/living/living.dm [hunk 5]
```diff
@@ -1074,13 +1074,17 @@
...
 	to_chat(src, span_danger("You try to remove [who]'s [what.name]..."))
 	log_message("[key_name(who)] is being stripped of [what] by [key_name(src)]", LOG_ATTACK, color="red")
 	who.log_message("[key_name(who)] is being stripped of [what] by [key_name(src)]", LOG_VICTIM, color="red", log_globally = FALSE)
 	what.add_fingerprint(src)
-	if(do_mob(src, who, what.strip_delay, interaction_key = what))
+	//SKYRAT EDIT ADDITION
+	//if(do_mob(src, who, what.strip_delay, interaction_key = what))
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/living.dm]:
```suggestion
	// SKYRAT EDIT ADDITION - Thieving gloves
	// if(do_mob(src, who, what.strip_delay, interaction_key = what))
```

File: code/modules/mob/living/living.dm [hunk 6]
```diff
@@ -1074,13 +1074,17 @@
...
+	if(do_mob(src, who, what.strip_delay * (HAS_TRAIT(who, TRAIT_STICKY_FINGERS) ? 0.5 : 1), interaction_key = what))
```
Comment (GoldenAlpharex) [inline code/modules/mob/living/living.dm:1089]: Those numbers should be defines rather than being hardcoded.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Few minor changes needed.

File: code/datums/elements/strippable.dm [hunk 7]
```diff
@@ -148,18 +151,20 @@
...
+			ignored_mobs = user,
+		)
 
 	to_chat(user, span_danger("You try to remove [source]'s [item]..."))
 	user.log_message("[key_name(source)] is being stripped of [item] by [key_name(user)]", LOG_ATTACK, color="red")
 	source.log_message("[key_name(source)] is being stripped of [item] by [key_name(user)]", LOG_VICTIM, color="red", log_globally=FALSE)
 	item.add_fingerprint(src)
 
-	if(ishuman(source))
+	if(ishuman(source) && !is_silent) //SKYRAT EDIT ADDITION original if(ishuman(source))
```
Comment (GoldenAlpharex) [inline code/datums/elements/strippable.dm]: Oh, also, for this, did you make sure that it still properly gets logged if they try to take something off of someone that's SSD in the logs? I'd assume so, but you're never too sure.

Comment (Seris02) [reply code/datums/elements/strippable.dm]: two lines above nerd

**Review by GoldenAlpharex — APPROVED** [review]
Code checks out!

**Comment by GoldenAlpharex** [comment]
As such, since it passes CI and the other one doesn't, and because we've taken some measures so that stuff like this doesn't happen again, I'll be merging this one, as it the one I believe has the highest chance of being maintained in the future.

**Pull request merged**